test/sql/types/timestamp/test_timestamp_tz.test

```diff
@@ -0,0 +1,12 @@
+# name: test/sql/types/timestamp/test_timestamp_tz.test
+# description: Test TIMESTAMP WITH TIME ZONE common operations
+# group: [timestamp]
+
+statement ok
+PRAGMA enable_verification
+
+# Cast from string
+query I
+select timestamptz '2021-11-15 02:30:00';
```

Comment:
Since the casts are just the same as for `TIMESTAMP` but just using a different result type, I don't think there is anything to do here - at least for ISO-8601. But yes, the ability to parse/cast dates in a local format would need hooks. I'm also not sure that we have the tables built for things like `MONTHNAME` (which is not implemented).

Reply:
I was under the impression timestamptz casts needed access to the list of timezones as well, no? e.g. in Postgres:

```sql
SELECT timestamptz '2021-01-01 19:00:00 US/Pacific';
--  2021-01-02 04:00:00+01
```

Reply:
Yeah, but as you said, that would require cast extensibility. So the current implementation just uses the existing casting logic.

Reply:
Future work then :)

Reply:
Yes! Although I would say that all we really want is `VARCHAR` => `TIMEZONETZ` For the other direction we might want to leave it as it is, but add a `DATE_FORMAT` function.